agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryChannel.java

```diff
@@ -86,6 +85,10 @@
   //  operationLogger?
   private final AtomicBoolean friendlyExceptionThrown = new AtomicBoolean();
 
+  private static final int MAX_STATSBEAT_ERROR_COUNT = 10;
+  private static final AtomicInteger statsbeatErrorCount = new AtomicInteger();
+  private static final AtomicBoolean atLeastOneStatsbeatSuccess = new AtomicBoolean();
```

Comment:
not using atomicity
```suggestion
  private static final volatile boolean atLeastOneStatsbeatSuccess;
```

Reply:
i prefer to keep AtomicBoolean since setting on volatile is not atomic. statsbeat is at least every 15 mins. there is no perf gain in this case.

Reply:
you aren't using atomicness though